### PR TITLE
Add Endpoins for Project Membership manipulation [SCI-5001]

### PR DESCRIPTION
**Pull request by okriuchykhin**
Jira ticket: [SCI-5001](https://biosistemika.atlassian.net/browse/SCI-5001)

### What was done
Add Endpoins for Project Membership manipulation

### CODE REVIEW

File: app/controllers/api/v1/user_projects_controller.rb [hunk 1]
```diff
@@ -23,11 +24,44 @@ def show
...
+        @user_project.role = user_project_params[:role]
+        return render body: nil, status: :no_content unless @user_project.changed?
+
+        @user_project.assigned_by = current_user
+        @user_project.save!
+        render jsonapi: @user_project, serializer: UserProjectSerializer, status: :ok
+      end
+
+      def destroy
+        @user_project.destroy!
```
Comment (radiokills) [inline app/controllers/api/v1/user_projects_controller.rb:45]: Should we check here if the user is not unassigning himself?

Comment (radiokills) [reply app/controllers/api/v1/user_projects_controller.rb:45]: Or at least check if he is owner (or the last user on a project).

**Review by radiokills — COMMENTED** [review]
Straight to the point! Good work. Should we check when removing the user?

File: app/controllers/api/v1/user_projects_controller.rb (hunk 1 above)
Comment (okriuchykhin) [reply app/controllers/api/v1/user_projects_controller.rb:45]: Do we need such checks in an API?

Comment (okriuchykhin) [reply app/controllers/api/v1/user_projects_controller.rb:45]: I think it is better to move such check to the model itself

Comment (radiokills) [reply app/controllers/api/v1/user_projects_controller.rb:45]: true

**Review by radiokills — APPROVED** [review]

**Pull request merged**